transformer_engine/jax/cpp_extensions.py

```diff
@@ -1904,12 +1904,22 @@ def abstract(qkv_aval, bias_aval, seqlen_or_cu_seqlen_aval, seed_aval, *, attn_b
         rng_state_shape = (seed_aval.shape[0], checker.rng_state_size)
         rng_state_aval = seed_aval.update(shape=rng_state_shape, dtype=checker.rng_state_dtype)
 
+        # NOTE: We can't rely on checking bias type here because we want to support arbitrary bias
+        #       shapes the bias tensor is being used as a workaround for an arbitrary mask
+        #       (True/False -> 0/-Inf) with bias type set to NO_BIAS.
```

Comment:
Could you help me understand in which case the bias_type is `NO_BIAS` but the bias_aval.shape is not zero?

Reply:
This is an incorrect comment. The arbitrary mask workaround is supported via `AttnBiasType.POST_SCALE_BIAS` and the code is implemented for it but I neglected to update the comment here.